nix/modules/nixos/holo-orchestrator.nix

```
@@ -0,0 +1,173 @@
+# Module to configure a machine as a holo-orchestrator.
+
+# blueprint specific first level argument that's referred to as "publisherArgs"
+{
+  inputs,
+  flake,
+  ...
+}:
+
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.holo.orchestrator;
+in
+{
+  options.holo.orchestrator = {
+    enable = lib.mkOption {
+      description = "enable holo-orchestrator";
+      default = true;
+    };
+
+    autoStart = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+    };
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = inputs.self.packages.${pkgs.stdenv.system}.rust-workspace;
+    };
+
+    hubAuthScriptPath =  lib.mkOption {
+      type = lib.types.path;
+      default = "${flake}/scripts/hub_auth_setup.sh";
+    };
+
+    rust = {
+      log = lib.mkOption {
+        type = lib.types.str;
+        default = "debug";
+      };
+
+      backtrace = lib.mkOption {
+        type = lib.types.str;
+        default = "1";
+      };
+    };
+
+    mongo = {
+      listenHost = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+
+      listenPort = lib.mkOption {
+        type = lib.types.int;
+        default = 27017;
+      };
+
+      url = lib.mkOption {
+        type = lib.types.str;
+        default = "${cfg.mongo.listenHost}:${builtins.toString cfg.mongo.listenPort}";
+      };
+    };
+
+    nats = {
+      listenHost = lib.mkOption {
+        type = lib.types.str;
+        default = "127.0.0.1";
+      };
+
+      listenPort = lib.mkOption {
+        type = lib.types.int;
+        default = 4222;
+      };
+
+      url = lib.mkOption {
+        type = lib.types.str;
+        default = "${cfg.nats.listenHost}:${builtins.toString cfg.nats.listenPort}";
+      };
+
+      tlsInsecure = lib.mkOption {
+        type = lib.types.bool;
+      };
+
+      nscPath = lib.mkOption {
+        type = lib.types.path;
+        default = "/var/lib/.local/share/nats/nsc";
+      };
+
+      sharedCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/shared_creds";
+      };
+
+      localCredsPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.nscPath}/local_creds";
+      };
+
+      rootAuthNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/AUTH_ROOT_SK.nk";
+      };
+
+      signingAuthNkeyPath = lib.mkOption {
+        type = lib.types.path;
+        default = "${cfg.nats.localCredsPath}/AUTH_SK.nk";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.mongodb = {
```

assuming you want to enable the existing nixpkgs module for mondodb, please look at these: https://search.nixos.org/options?channel=24.11&from=0&size=50&sort=relevance&type=packages&query=services.mongodb

```suggestion
    services.mongodb = {
```